luigi/worker.py

```diff
@@ -1127,8 +1132,16 @@ def _keep_alive(self, get_work_response):
             return get_work_response.n_pending_last_scheduled > 0
         elif self._config.count_uniques:
             return get_work_response.n_unique_pending > 0
+        elif get_work_response.n_pending_tasks <= 0:
+            return False
+        elif not self._config.keep_alive_ttl:
```

Comment:
does the default value drop you into this elif block?

Reply:
Yes! Yes, it does.